packages/mask/src/components/shared/PersonaBoundary.tsx

```diff
@@ -0,0 +1,96 @@
+import { Button, Stack } from '@mui/material'
+import { memo, ReactNode, useCallback, useMemo } from 'react'
+import { makeStyles } from '@masknet/theme'
+import type { PluginId } from '@masknet/plugin-infra'
+import { PersonaConnectStatus, useCurrentPersonaConnectStatus } from '../DataSource/usePersonaConnectStatus'
+import { useI18N } from '../../utils'
+import { Icons } from '@masknet/icons'
+
+const useStyles = makeStyles()((theme) => ({
+    root: {
+        position: 'relative',
+    },
+    mask: {
+        position: 'absolute',
+        top: 0,
+        bottom: 0,
+        left: 0,
+        right: 0,
+        cursor: 'pointer',
+    },
+    button: {
+        borderRadius: '99px',
+        backgroundColor: theme.palette.maskColor.dark,
+        color: '#fff',
+        marginTop: 'auto',
+        ':hover': {
+            color: 'fff',
+            backgroundColor: theme.palette.maskColor.dark,
+        },
+    },
+}))
+
+type SupportChildren = ((status: PersonaConnectStatus) => ReactNode) | ReactNode
+
+interface PersonaBoundaryProps {
+    handlerPosition?: 'center' | 'top-right'
+    directTo?: PluginId
+    customHint?: boolean
+    children: SupportChildren
+    enableVerify?: boolean
+    beforeVerify?: () => void | Promise<void>
+}
+
+export const PersonaBoundary = memo<PersonaBoundaryProps>(
```

Comment:
`PersonaBoundary` is not specific enough. Some candidates for replacing it:

+ `ConnectPersonaBoundary`
+ `ConnectNextIDBoundary`